### DISCUSSION TITLE
SimpleLinkScraper's HTML Page Limit of 1 MB Too Low?

### DISCUSSION THREAD

**Discussion by michaelnordmeyer** [MarginaliaSearch/MarginaliaSearch#137] [General]
MarginaliaSearch's [`SimpleLinkScraper` limits HTML pages to a size of 1 MB](https://github.com/MarginaliaSearch/MarginaliaSearch/blob/927bc0b63c131a108293eac27cd500a199ac3c01/code/processes/live-crawling-process/java/nu/marginalia/livecrawler/SimpleLinkScraper.java#L193).

If I grab the single-page HTML version of [Tolstoi's *War and Peace* from Project Gutenberg](https://www.gutenberg.org/ebooks/2600), it's 3.9 MB in size.

Of course, an ordinary HTML page is not a book, but I wonder how many pages get dismissed because of this size restriction and how many of them should be indexed. Are there any statistics?

**Comment by vlofgren** [comment]
It's only the SimpleLinkScraper used for the real time crawler that has this behavior.   

The big boy crawler will [fetch up to 10 MB](https://github.com/MarginaliaSearch/MarginaliaSearch/blob/0d59202aca890a8c2c0a0d2b2286f7a5e879ae3d/code/processes/crawling-process/java/nu/marginalia/crawl/fetcher/warc/WarcRecorder.java#L38), but [truncates at that point](https://github.com/MarginaliaSearch/MarginaliaSearch/blob/0d59202aca890a8c2c0a0d2b2286f7a5e879ae3d/code/processes/crawling-process/java/nu/marginalia/crawl/fetcher/warc/WarcInputBuffer.java#L82) rather than rejecting the data altogether.  

I don't there's much discussion that this break-point should probably be the same across the board, and should also probably be configurable and not hard coded, so I fixed that.

---

Though this made me realize there's a bigger issue, the code that's responsible for breaking text into sentences has an [upper limit on how many sentences it will extract](https://github.com/MarginaliaSearch/MarginaliaSearch/blob/bae44497fe030243fe506e1dbf9bc98a55e5efe6/code/libraries/language-processing/java/nu/marginalia/language/sentence/SentenceExtractor.java#L46), and that's relatively low, 1000.   

On the one hand, if we take War and Peace as an example, if you remove that cap it breaks down into 30,717 sentences, so there's clearly a lot of that text that wouldn't be searchable.  Though on the other hand, the [wikipedia article for Aristotle](https://en.wikipedia.org/wiki/Aristotle) is about 450 sentences using this logic, so it seems relatively safe to say this will only affect serious outliers in terms of length.

There's also performance implications in completely removing this limit, but it's worth investigating increasing it a bit, as it may be set a bit low.  I'll see if I can get some stats on how often this length cap is actually reached, and some examples of what such documents look like.
